src/Builder/Visitor/AddPluginToPluginCollectionExtendContainerVisitor.php

```diff
@@ -134,9 +175,51 @@ protected function handleContainerExtendClosure(Closure $closure, array $addPlug
             $newMethodCall = (new BuilderFactory())
                 ->methodCall($addPluginCalls[0]->var, $addPluginCalls[0]->name, $arguments);
 
-            array_splice($closure->stmts, $newPluginAddCallIndex, 0, [new Expression($newMethodCall)]);
+            array_splice($closure->stmts, (int)$newPluginAddCallIndex, 0, [new Expression($newMethodCall)]);
         }
 
         return $closure;
     }
+
+    /**
+     * @param \PhpParser\Node $node
+     *
+     * @return array<string>
+     */
+    protected function getPluginList(Node $node): array
+    {
+        $plugins = [];
+
+        foreach ($node->stmts as $stmt) {
+            if (
+                $stmt instanceof Expression
+                && $stmt->expr instanceof MethodCall
+                && count($stmt->expr->args) >= 2
+                && $stmt->expr->args[1]->value instanceof Closure
+            ) {
```

Comment:
Usually if you have such long if conditions, it's also better for readability to extract them into new methods with meaningful names. For example in your case you can use:

```
protected function isExpressionWithClosure(<object class> $stmt): bool
{
    return $stmt instanceof Expression
                && $stmt->expr instanceof MethodCall
                && count($stmt->expr->args) >= 2
                && $stmt->expr->args[1]->value instanceof Closure;
}
```

and here you can just use the method call and maybe also with early return:

```
foreach ($node->stmts as $stmt) {
   if (!$this->isExpressionWithClosure($stmt)) {
      continue;
   }
   
   /** @var \PhpParser\Node\Expr\Closure $closure */
   $closure = $stmt->expr->args[1]->value;
   ...
}
```

Reply:
Good idea 
Added isExpressionWithClosure